bin/kano-settings-onboot

```diff
@@ -360,172 +158,132 @@ def ensure_correct_volume_levels():
+    override_models(edid, model)
+    calculate_is_monitor(edid)
 
-        if not force_mode and (not has_config_comment('kano_screen_used') or
-                                   get_config_comment('kano_screen_used', model)):
+    # Fix HDMI audio status.
+    if not edid['hdmi_audio'] and is_HDMI():
+        logger.info('HDMI audio not supported on this screen, changing to analogue.')
+        if not dry_run:
+            set_to_HDMI(False)
 
-            # The screen is either the same or not set at all
-            logger.info('Explicit HDMI configuration detected, exiting.')
-            sys.exit(1)
-        else:
-            logger.info('New screen was detected, reconfiguring...')
+    # Calculate changes needed...
+    changes = compare_and_set_full_range(edid, status, model, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    # override modes, calculate is_monitor
-    override_models(edid, model)
-    calculate_is_monitor(edid)
+    changes = compare_and_set_overscan(edid, status, model, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    if verbose:
-        # Explain the new screen settings
-        print 'New status:', json.dumps(status, indent=4)
-        print 'New EDID:', json.dumps(edid, indent=4)
+    changes = compare_and_set_optimal_resolution(edid, status, supported_modes, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    # fix hdmi audio status
-    if not edid['hdmi_audio'] and is_HDMI():
-        logger.info('hdmi audio not supported on this screen, changing to analogue')
-        set_to_HDMI(False)
+    return (reboot_required, RC_SUCCESS)
 
-    # calculate changes needed
-    overall_changes = False
 
-    changes = compare_and_set_mode(edid, status)
-    overall_changes = overall_changes or changes
+def calculate_is_monitor(edid):
+    """
+    TODO: What is this for?
+    """
+    edid['target_full_range'] = edid['is_monitor']
 
-    changes = compare_and_set_full_range(edid, status)
-    overall_changes = overall_changes or changes
 
-    changes = compare_and_set_overscan(edid, status)
-    overall_changes = overall_changes or changes
+def reboot(dry_run=False):
+    """
+    Flush logs to disk and reboot the kit.
+    """
+    if not dry_run:
+        logger.sync()
+        run_cmd('kano-checked-reboot changes systemctl reboot --force')
 
-    changes = set_resolution_for_screen_kit(supported) or \
-              compare_and_set_optimal_resolution(edid, status, supported, verbose)
-    overall_changes = overall_changes or changes
 
-    if overall_changes:
-        if verbose:
-            print 'changes applied:', changes
+if __name__ == '__main__':
+    logger.force_log_level('info')
 
-        # write comment to config
-        set_config_comment('kano_screen_used', model)
-        end_config_transaction()
+    if os.geteuid() != 0:
```

Comment:
not really that important, but I think `getuid` is safer and clearer in this case.
We won't really use privilege escalation through `setuid` bit no?

Reply:
Reading back on it, it's definitely clearer as you suggest. Thanks for catching that 👍